view/frontend/web/js/react-src/src/index.tsx

```diff
@@ -14,13 +14,16 @@ export const renderReactElement = (container: Element): void => {
       if (cartForm == null) throw new Error('Product form not found')
       const formURL = cartForm.getAttribute('action') ?? window.location.href
       if (!formURL) throw new Error('Product form URL not found')
-      await fetch(formURL, {
+      const response = await fetch(formURL, {
         method: 'POST',
         body: new FormData(cartForm as HTMLFormElement),
         headers: {
           'X-Requested-With': 'XMLHttpRequest'
         }
       })
+
+      const data = await response.json()
+      if (data.error) data.backUrl ? window.location.href = data.backUrl : window.location.reload()
```

Comment:
instead of reloading the window, if this promise throws a message, the error message will be displayed when `amwalPrePayTriggerError` is fired to the dialog and the modal will be closed. I think this will be a better user experience.

Reply:
I tried this initially but this wasn't working as expected. The modal would get stuck and no error message was displayed